From b60ba7a5514bd313f19b8340e24cc200ffc5ff70 Mon Sep 17 00:00:00 2001
From: Olivier Mehani <olivier.mehani@nicta.com.au>
Date: Sun, 25 Apr 2010 17:12:08 +1000
Subject: [PATCH 1/6] Don't open console on --bkeymap

Check that -b hasn't specified, similarly to -m, before loadkeys tries to open
the console. As for --mktable, --bkeymap doesn't need it. This allows to dump
binary keymaps in a larger range of situations where ownership of the console
device is not guarranteed e.g., when running from an Xterm.

Signed-off-by: Olivier Mehani <olivier.mehani@nicta.com.au>
Signed-off-by: Alexey Gladkov <gladkov.alexey@gmail.com>
---
 src/loadkeys.c |    2 +-
 src/loadkeys.y |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/loadkeys.c b/src/loadkeys.c
index a827f8e..a610722 100644
--- a/src/loadkeys.c
+++ b/src/loadkeys.c
@@ -2090,7 +2090,7 @@ main(int argc, char *argv[]) {
 	}
 
 	prefer_unicode = optu;
-	if (!optm) {
+	if (!optm && !optb) {
 		/* check whether the keyboard is in Unicode mode */
 		fd = getfd(NULL);
 		if (ioctl(fd, KDGKBMODE, &kbd_mode)) {
diff --git a/src/loadkeys.y b/src/loadkeys.y
index 03654ac..7df9d04 100644
--- a/src/loadkeys.y
+++ b/src/loadkeys.y
@@ -367,7 +367,7 @@ main(int argc, char *argv[]) {
 	}
 
 	prefer_unicode = optu;
-	if (!optm) {
+	if (!optm && !optb) {
 		/* check whether the keyboard is in Unicode mode */
 		fd = getfd(NULL);
 		if (ioctl(fd, KDGKBMODE, &kbd_mode)) {
-- 
1.7.3.4

